modules/libcom/src/calc/postfix.c

```diff
@@ -27,6 +27,10 @@
 #include "postfixPvt.h"
 #include "shareLib.h"
 
+#ifdef RTEMS_HAS_ALTIVEC
+#pragma GCC push_options
+#pragma GCC optimize ("O0")
```

Comment:
I fully agree.  I have similar thoughts about the MSVC pragma in 'calcPerform.c'.  I can't help but wonder at the similar structure of these two pieces of code.

Reply:
Not even with a HW-debugger I had unfortunately not managed to analyse what was wrong at that point. 
The simplest calcTest fails. I suspect a problem with cache/FPU etc...

Reply:
FYI.  I checked and PPC/altivec load/store instructions do need 16-bit alignment.  It might be worth trying to replace `("O0")` with `("no-altivec")`.

Some things maybe useful to check.

1. Look for any 'vec_*' instructions in the disassembly for postfix.o
2. See if memcpy() calls select the libc function, or the GCC builtin.

Reply:
"no-altivec" does not help:
***** epicsCalcTest *****
1..630
ok  1 - 0
ok  2 - 1
ok  3 - 2
ok  4 - 3
ok  fatal source: RTEMS_FATAL_SOURCE_EXCEPTION
exception vector 6 (0x6)
  next PC or address of fault = 0x000461e0
  saved MSR = 0x0200b032
  context = task, ISR nest level = 0
  thread dispatch disable level = 0
  R0  = 0x00046190 R1  = 0x002c5a00 R2  = 0x00000000 R3  = 0x00000000
  R4  = 0x002225dc R5  = 0x00000001 R6  = 0x00000001 R7  = 0x002010e0
  R8  = 0x00000000 R9  = 0x00000001 R10 = 0x80000000 R11 = 0x00ada1b0
  R12 = 0x22000444 R13 = 0x00277990 R14 = 0x00210e44 R15 = 0x00210e74
  R16 = 0x00000001 R17 = 0x002225dc R18 = 0x00211404 R19 = 0x00211408
  R20 = 0x00210000 R21 = 0x00000000 R22 = 0x00211214 R23 = 0x00210e44
  R24 = 0x002c5a08 R25 = 0x002c5ff0 R26 = 0x00ad9f28 R27 = 0x00232dc5
  R28 = 0x00000001 R29 = 0x00ad9f31 R30 = 0x002c5a08 R31 = 0x002010e2
  CR  = 0x22000444
  CTR = 0x00045f38
  XER = 0x00000000
  LR  = 0x00046190
  DAR = 0x00ad9f29
  executing thread ID = 0x0b010001, name =
Stack Trace:
  IP: 0x000461e0, LR: 0x00046190
--^ 0x000037f4--^ 0x000040ac--^ 0x00057560--^ 0x0002e214--^ 0x00040254
--^ 0x0006e548--^ 0x0007e144--^ 0x0007d1dc--^ 0x0007d12c

I set it to "O0" again.